sample/payments/CreatePaymentUsingCheckoutJs.php

```diff
@@ -0,0 +1,152 @@
+    ->setTax(1.3)
+    ->setSubtotal(17.50);
+
+// ### Amount
+// Lets you specify a payment amount.
+// You can also specify additional details
+// such as shipping, tax.
+$amount = new Amount();
+$amount->setCurrency("USD")
+    ->setTotal(20)
+    ->setDetails($details);
+
+// ### Transaction
+// A transaction defines the contract of a
+// payment - what is the payment for and who
+// is fulfilling it. 
+$transaction = new Transaction();
+$transaction->setAmount($amount)
+    ->setItemList($itemList)
+    ->setDescription("Payment description")
+    ->setInvoiceNumber(uniqid());
+
+// ### Redirect urls
+// Set the urls that the buyer must be redirected to after 
+// payment approval/ cancellation.
+$baseUrl = getBaseUrl();
+$redirectUrls = new RedirectUrls();
+$redirectUrls->setReturnUrl("$baseUrl/ExecutePayment.php?success=true")
+    ->setCancelUrl("$baseUrl/ExecutePayment.php?success=false");
+
+// ### Payment
+// A Payment Resource; create one using
+// the above types and intent set to 'sale'
+$payment = new Payment();
+$payment->setIntent("sale")
+    ->setPayer($payer)
+    ->setRedirectUrls($redirectUrls)
+    ->setTransactions(array($transaction));
+
+
+// For Sample Purposes Only.
+$request = clone $payment;
+
+// ### Create Payment
+// Create a payment by calling the 'create' method
+// passing it a valid apiContext.
+// (See bootstrap.php for more on `ApiContext`)
+// The return object contains the state and the
+// url to which the buyer must be redirected to
+// for payment approval
+try {
+    $payment->create($apiContext);
+} catch (Exception $ex) {
+    // NOTE: PLEASE DO NOT USE RESULTPRINTER CLASS IN YOUR ORIGINAL CODE. FOR SAMPLE ONLY
+    ResultPrinter::printError("Created Payment Using PayPal. Please visit the URL to Approve.", "Payment", null, $request, $ex);
+    exit(1);
+}
+
+// ### Get redirect url
+// The API response provides the url that you must redirect
+// the buyer to. Retrieve the url from the $payment->getApprovalLink()
+// method
+$approvalUrl = $payment->getApprovalLink();
+
+
+// Checkout JS https://developer.paypal.com/demo/checkout/#/pattern/server
+// NOTE: The checkout.js script should be added to your head element not directly to the body as shown in the sample here.
+echo '<script src="https://www.paypalobjects.com/api/checkout.js"></script>';
+
+// NOTE: As an Alternative you can make a request with paypal.request.post to a page that returns the Payment ID value
+echo "<script>
+paypal.Button.render({
+
+    env: 'sandbox', // sandbox | production
+
+    // Show the buyer a 'Pay Now' button in the checkout flow
+    commit: true,
+
+    // payment() is called when the button is clicked
+    payment: function() {
+        // Return the Payment ID
+        return '$payment->id';
+    },
+
+    // onAuthorize() is called when the buyer approves the payment
+    onAuthorize: function(data, actions) {
+        // Redirect to existing Execute Payment
+        return actions.redirect();
```

Comment:
Are there other options than doing a redirect ? Can I fetch all the relevant field here, and make a normal HTTP post call to my server to do an execute for me ? It would be nice to mention the most common flow used.

Reply:
This is the simplest way with the existing sample set up, I plan on creating another example with Ajax calls using the PayPal request object that is built into the Checkout library.

Reply:
I think we should just stick to one. Let people do other's from documentation